Removed compatibility issues from PomdpX and test_PomdpX

khalibartan opened the pull request:
In TestPomdpXReaderString io.StringIO is replaced by six.StringIO because in py2 io.StringIO doesn't read string inputs.
Also in PomdpXWriter class and TestPomdpXWriter class etree is being import form lxml or xml.etree.ElementTree because cElementTree is having some problems in taking dictionary attributes.
check issue #493 #448

khalibartan committed:
replace xml.etree.cElementTree by xml.etree.ElementTree and io.StringIO by six.StringIO

khalibartan committed:
Removed compatability issues from ProbModelXML.py and its test

khalibartan commented:
@ankurankan for `test_ProbModelXML`, the fix isn't good. I have replace some strings by unicode in class `TestProbModelXMLReaderString`, method `test_potential` ,dictionary `potential_expected` because these strings were not being converted to unicode. 
Suggest some better solution

khalibartan commented:
Should I replace print by six.print_ because this landscape bot is showing repository health decreased.
I don't think that's an issue?

ankurankan commented:
@khalibartan I think you will need to specify strings to be unicode in case of python2. A better way to do this rather to append `u` in front of strings is to add `# -*- coding: utf-8 -*-` at the top of the file.
Don't worry too much about landscape.

khalibartan commented:
`# -*- coding: utf-8 -*-` is there in  test_ProbModelXML.py .But it wasn't working as intended. Not all strings should be converted to unicode only some of the strings in `potential_expected` should be converted.

khalibartan committed:
commented out the import of cElementTree as not expected behaviour

khalibartan commented:
@ankurankan  can take a look at the pull request. Anything that needs to be changed ?

khalibartan commented:
Don't merge these commits. I think we should use cElementTree in testing and ElementTree in normal because only problem with cElementTree is that it is not taking dictionary arguments but it is faster.

khalibartan committed:
Fixed issues with XMLBeliefNetwork and its test

khalibartan commented:
@ankurankan can give views on the commits if something needs to be done?

khalibartan committed:
Removed compatibility issue from XMLBIF & test

xml.etree.cElementTree was not working with dictonary arguments.

khalibartan committed:
Removed compatibility issues from GibbsSampling

The original author has changed the value of var in
_get_kernel_from_bayseian_model which caused grade to be skipped.

ankurankan commented on pgmpy/inference/Sampling.py:
```diff
@@ -257,10 +257,10 @@ def _get_kernel_from_bayesian_model(self, model):
 
-        for var in self.variables:
-            other_vars = [v for v in self.variables if var != v]
+        for Var in self.variables:
```
@khalibartan Why was this failing earlier?

khalibartan replied:
if look down below the person who wrote this was using var in a for loop also. Which caused some values to skipped in self.transition_models.See line 268.

ankurankan commented:
@khalibartan I think except for MPLP everything else is fixed ? 
And if I remember correctly there was some issue in the algorithm of MPLP. So if you don't want to go through the algorithm, leave that and I will merge this ?

khalibartan commented:
@ankurankan Yes apart from mplp all compatibility issues have been fixed in this pull request.You can merge this one. 
 Also I want to go for mplp .I have just began working on mplp and I'm just about to find the origin of the bug. I have some good lead on it.

ankurankan commented:
@khalibartan Cool.. No problem.

khalibartan committed:
Removed Compatibility issues from mplp and its test

Through backtracking the error I found that there was issue with
decoded_result_assignment which gave hint that self.objective was having
some issue.After printing the state of self.objective in py2 and py3 i
realized that assignment of self.objective was somewhere wrong. Printed
the values of assignment in _update_message and found that updating was
wrong . After a some while realized that division in py2 doesn't give
float if numbers are integers. (see updation of phi in _update_message).

khalibartan commented:
@ankurankan look at the latest commit. mplp is fixed . Now you can merge this pr..
